docs/tutorial.md

```diff
@@ -3,200 +3,457 @@
+- A string is a piece of text that can contain multiple characters. Strings are defined by using the quotes `''`
+- An array is a type that can hold multiple variables inside itself. Arrays are defined by using the square brackets `[]`
+- Object are basically advanced arrays, you will learn more about it later in this tutorial. Their defined by curly brackets `{}` 
+- Functions are also their own type.
+- A boolean is a type that can only be `true` or `false`
+- When something is not (yet) defined, its type is `undefined`
 
-Examples :
+```js
+var number = 1                  // Number type
+var string = 'This is a string' // String type
+var array = [1, 2, 3]           // Array type
+var object = {}                 // Object type
+var function = Function() {}    // Function type
+var boolean = true              // Boolean type
+var nothing                     // Undefined type
+```
 
-#### Echo Bot
+#### If-statements
 
-Here we're listening for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and resend the message.
-[`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event is only emitted when a player chats publicly.
+Sometimes you want to do different things based on a certain condition.  
+This can be achieved using if-statements.
 
 ```js
-function echo (username, message) {
-  if (username !== bot.username) {
-    bot.chat(message)
-  }
-}
+var name = 'Bob'
 
-bot.on('chat', echo) // Only emitted when a player chats publicly.
+if (name == 'Bob') {
+  console.log('Your name is Bob')
+}
+else if (name == 'Alice') {
+  console.log('Your name is Alice')
+}
+else {
+  console.log('Your name is not Bob or Alice')
+}
 ```
 
-`bot.on('chat', echo)` - will listen for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and execute `echo` function with arguments `username`, `message`, `translate`, `jsonMsg`, and `matches` when the event emitted.
+An if-statement is created using the `if` keyword. After that you have a condition between the round brackets `()` followed by the body between the curly brackets `{}`
+A condition has to be something that computes to a boolean.  
+In this case it uses an equal operator `==` which will be `true` if the value in front is the same as the value after. Otherwise it will be `false`
+If the condition is `true` the code in the body will be executed.  
+You can chain an if-statement with an if-else-statement or an else-statement.  
+You can have multiple if-else-statements, but there can't be any code between the statements.  
+If you have an else-statement, it will be called only if all the chained statements before it are `false`
 
-#### Welcome Bot
+```js
+if (true) {
+  ...
+}
+console.log(5); // This is invalid.
+else {
+  ...
+}
+```
+
+#### Loops
```

Comment:
Maybe, add something about `for` loops ?

Reply:
Done!

Reply:
You could also add the other variants of for loops:
```js
for (const index in myarray) { // iterate over the keys (indices) of myarray
  const item = myarray[index]
  // for each item in myarray
}
for (const item of myarray) { // iterate directly over the elements of myarray
  // for each item in myarray
}
```

Reply:
Done, although I feel it's a bit vague and hard to explain easily.